Subject: Add a helpful header to a new crontab
Author: Javier Fernández-Sanguino Peña <jfs@debian.org>
Last-Update: 2010-04-11

When creating a new crontab, seed it with helpful instructions (as comments) on
how to create an entry.

Acked-by: Christian Kastner <debian@kvr.at>
Index: patched/crontab.c
===================================================================
--- patched.orig/crontab.c	2010-05-06 18:18:57.090560587 +0200
+++ patched/crontab.c	2010-05-06 18:18:57.762561131 +0200
@@ -549,6 +549,7 @@
 	WAIT_T		waiter;
 	PID_T		pid, xpid;
 	mode_t		um;
+	int             add_help_text = 0;
 
 	log_it(RealUser, Pid, "BEGIN EDIT", User);
 	(void) snprintf(n, MAX_FNAME, CRON_TAB(User));
@@ -563,6 +564,7 @@
 			perror("/dev/null");
 			exit(ERROR_EXIT);
 		}
+		add_help_text = 1;
 	}
 
 	um = umask(077);
@@ -581,6 +583,33 @@
 
 	Set_LineNum(1)
 
+	if (add_help_text) {
+		fprintf(NewCrontab, 
+"# Edit this file to introduce tasks to be run by cron.\n"
+"# \n"
+"# Each task to run has to be defined through a single line\n"
+"# indicating with different fields when the task will be run\n"
+"# and what command to run for the task\n"
+"# \n"
+"# To define the time you can provide concrete values for\n"
+"# minute (m), hour (h), day of month (dom), month (mon),\n"
+"# and day of week (dow) or use '*' in these fields (for 'any')."
+"# \n"
+"# Notice that tasks will be started based on the cron's system\n"
+"# daemon's notion of time and timezones.\n"
+"# \n"
+"# Output of the crontab jobs (including errors) is sent through\n"
+"# email to the user the crontab file belongs to (unless redirected).\n"
+"# \n"
+"# For example, you can run a backup of all your user accounts\n"
+"# at 5 a.m every week with:\n"
+"# 0 5 * * 1 tar -zcf /var/backups/home.tgz /home/\n"
+"# \n"
+"# For more information see the manual pages of crontab(5) and cron(8)\n" 
+"# \n"
+"# m h  dom mon dow   command\n" );
+	}
+
 	/* ignore the top few comments since we probably put them there.
 	 */
 	for (x = 0;  x < NHEADER_LINES;  x++) {
